### PR TITLE
Remove unhelpful try/catch statements

### PR DESCRIPTION
**Pull request by reidrankin**
This is based on #381.

There are a lot of unhelpful try/catch statements in the library (generally of the form `try { } catch (e) { return Promise.reject(e) }`) which don't actually do anything. This gets rid of them.

(This is a part of the changeset currently known as #376; I'm working on factoring out some of its parts that are separable concerns and make the diffs hard to read if all rolled into one thing.)

### CODE REVIEW

File: packages/xchain-litecoin/src/utils.ts [hunk 1]
```diff
@@ -120,7 +120,7 @@ export const getBalance = async (params: AddressParams): Promise<Balance[]> => {
       },
     ]
   } catch (error) {
-    return Promise.reject(new Error('Invalid address'))
+    throw new Error('Invalid address')
```
Comment (veado) [inline packages/xchain-litecoin/src/utils.ts:123]:
```ts
Error('Could not get balances for address ${params.address}') // ...not sure what the address is in `AddressParams`
```

**Review by veado — APPROVED** [review]
LGTM - just found some "typos" in few messages (mostly based on code we already had before)

Changes for `xchain-polkadot` have not been reviewed - we can ignore this package for now ...

File: packages/xchain-bitcoincash/src/client.ts [hunk 2]
```diff
@@ -232,27 +225,21 @@ class Client extends BaseXChainClient implements BitcoinCashClient, XChainClient
...
-    } catch (error) {
-      return Promise.reject(error)
+    const account = await getAccount({ haskoinUrl: this.getHaskoinURL(), address })
+    const txs = await getTransactions({
+      haskoinUrl: this.getHaskoinURL(),
+      address,
+      params: { offset, limit },
+    })
+
+    if (!account || !txs) throw new Error('Invalid address')
```
[the remark this answers is not in the document]
Comment (reidrankin) [reply packages/xchain-bitcoincash/src/client.ts:238]:
```suggestion
    if (!account) throw new Error(`Invalid address: ${address}`)
    if (!txs) throw new Error(`Transactions could not found for address ${address}`)
```

File: packages/xchain-bitcoincash/src/haskoin-api.ts [hunk 3]
```diff
@@ -31,19 +33,9 @@ const isErrorResponse = (response: any): boolean => {
...
-    if (!result) {
-      throw new Error('failed to query account by a given address')
-    }
-
-    return result
-  } catch (error) {
-    return Promise.reject(error)
-  }
+  const result: AddressBalance | ErrorResponse = (await axios.get(`${haskoinUrl}/address/${address}/balance`)).data
+  if (!result || isErrorResponse(result)) throw new Error('failed to query account by a given address')
```
[the remark this answers is not in the document]
Comment (reidrankin) [reply packages/xchain-bitcoincash/src/haskoin-api.ts:37]:
```suggestion
  if (!result || isErrorResponse(result)) throw new Error(`failed to query account by given address ${address}`)
```

File: packages/xchain-bitcoincash/src/haskoin-api.ts [hunk 4]
```diff
@@ -56,19 +48,9 @@ export const getAccount = async ({ haskoinUrl, address }: AddressParams): Promis
...
-    if (!result) {
-      throw new Error('failed to query transaction by a given hash')
-    }
-
-    return result
-  } catch (error) {
-    return Promise.reject(error)
-  }
+  const result: Transaction | ErrorResponse = (await axios.get(`${haskoinUrl}/transaction/${txId}`)).data
+  if (!result || isErrorResponse(result)) throw new Error('failed to query transaction by a given hash')
```
[the remark this answers is not in the document]
Comment (reidrankin) [reply packages/xchain-bitcoincash/src/haskoin-api.ts:52]:
```suggestion
  if (!result || isErrorResponse(result)) throw new Error(`failed to query transaction by a given hash ${txId}`)
```

File: packages/xchain-bitcoincash/src/haskoin-api.ts [hunk 5]
```diff
@@ -81,19 +63,9 @@ export const getTransaction = async ({ haskoinUrl, txId }: TxHashParams): Promis
...
-    if (!result) {
-      throw new Error('failed to query transaction by a given hash')
-    }
-
-    return result.result
-  } catch (error) {
-    return Promise.reject(error)
-  }
+  const result: RawTransaction | ErrorResponse = (await axios.get(`${haskoinUrl}/transaction/${txId}/raw`)).data
+  if (!result || isErrorResponse(result)) throw new Error('failed to query transaction by a given hash')
```
[the remark this answers is not in the document]
Comment (reidrankin) [reply packages/xchain-bitcoincash/src/haskoin-api.ts:67]:
```suggestion
  if (!result || isErrorResponse(result)) throw new Error(`failed to query transaction by a given hash ${txId}`)
```

File: packages/xchain-bitcoincash/src/node-api.ts [hunk 6]
```diff
@@ -10,27 +10,23 @@ import { TxBroadcastResponse, TxBroadcastParams } from './types'
...
+        method: 'sendrawtransaction',
+        params: [txHex],
+        id: uniqid(),
+      },
+      {
+        auth,
+      },
+    )
+  ).data
+  if (response.error) {
```
[the remark this answers is not in the document]
Comment (reidrankin) [reply packages/xchain-bitcoincash/src/node-api.ts:27]:
```suggestion
  if (isErrorResponse(response)) {
```

File: packages/xchain-bitcoincash/src/utils.ts [hunk 7]
```diff
@@ -72,25 +72,19 @@ export function getFee(inputs: number, feeRate: FeeRate, data: Buffer | null = n
...
-      {
-        asset: AssetBCH,
-        amount: baseAmount(confirmed.amount().plus(unconfirmed.amount()), BCH_DECIMAL),
-      },
-    ]
-  } catch (error) {
-    return Promise.reject(new Error('Invalid address'))
-  }
+  const account = await getAccount(params)
+  if (!account) throw new Error('No bchBalance found')
```
[the remark this answers is not in the document]
Comment (reidrankin) [reply packages/xchain-bitcoincash/src/utils.ts:76]:
```suggestion
  if (!account) throw new Error('BCH balance not found')
```

File: packages/xchain-litecoin/src/utils.ts (hunk 1 above)
Comment (reidrankin) [reply packages/xchain-litecoin/src/utils.ts:123]: yep, it's `.address`
```suggestion
    throw new Error(`Could not get balances for address ${params.address}`)
```

File: packages/xchain-bitcoincash/src/node-api.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (reidrankin) [reply packages/xchain-bitcoincash/src/node-api.ts:27]: Unfortunately, `ErrorResponse`/`isErrorResponse` are specifically for the Haskoin API; the specific field `error` just happens to be named identically.

**Pull request merged**